docker/Dockerfile

```
@@ -0,0 +1,291 @@
+
+FROM $BASE AS python_base_config
+ENV PYENV_ROOT="/opt/pyenv"
+ENV VIRTUAL_ENV=/opt/venv
+ENV PATH="$VIRTUAL_ENV/bin:$PYENV_ROOT/bin:$PYENV_ROOT/shims:$PATH"
+ENV LD_LIBRARY_PATH="$VIRTUAL_ENV/lib:$LD_LIBRARY_PATH"
+
+FROM python_base_config AS python_builder
+ARG PYTHON_VERSION
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update --quiet; apt-get install -y --no-install-recommends \
+        make \
+        build-essential \
+        libssl-dev \
+        libbz2-dev \
+        libreadline-dev \
+        libsqlite3-dev \
+        wget \
+        curl \
+        llvm \
+        libncurses5-dev \
+        xz-utils \
+        tk-dev \
+        libxml2-dev \
+        libxmlsec1-dev \
+        libffi-dev \
+        liblzma-dev \
+        git \
+        ca-certificates \
+     && rm -rf /var/lib/apt/lists/*
+RUN curl -L \
+    https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer \
+    | bash
+RUN CFLAGS="-O2 -pipe" \
+    CONFIGURE_OPTS="--enable-shared --with-computed-gotos" \
+    # --enable-optimizations # possible performance improvements if building for local use
+    pyenv install -v $PYTHON_VERSION && \
+    pyenv global $PYTHON_VERSION
+RUN python -m venv $VIRTUAL_ENV && pip install -U pip && pip install -U setuptools
+
+
+# Assemble the base image that will be used to build the final base runtime image
+
+FROM python_base_config AS host
+COPY --from=python_builder $PYENV_ROOT $PYENV_ROOT
+COPY --from=python_builder $VIRTUAL_ENV $VIRTUAL_ENV
+COPY --from=oneapi_base_toolkit_installer $ONEAPI_INSTALL_DIR $ONEAPI_INSTALL_DIR
+# Those environment variables are required by the dpnp runtime
+ENV OCL_ICD_FILENAMES_RESET=1
+ENV OCL_ICD_FILENAMES=libintelocl.so
+
+
+# Install an up-to-date cmake that is necessary for some builds
+# and also some common prerequisites for dpctl, dpnp and numba-dpex builds
+
+FROM host AS base_builder
+ARG TMPDIR
+ARG CMAKE_VERSION
+ARG CMAKE_VERSION_BUILD
+ARG CMAKE_BUILD_DIR=$TMPDIR/cmake
+ARG INTEL_PYPI_URL
+ARG INTEL_NUMPY_VERSION
+ARG INTEL_SCIPY_VERSION
+RUN  apt-get update --quiet \
+    && apt-get install -y --no-install-recommends \
+        build-essential \
+        libtool \
+        autoconf \
+        unzip \
+        libssl-dev \
+        wget \
+        ninja-build \
+    && rm -rf /var/lib/apt/lists/* \
+    && apt purge -y --auto-remove cmake \
+    && mkdir -p $CMAKE_BUILD_DIR \
+    && cd $CMAKE_BUILD_DIR \
+    && wget https://cmake.org/files/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}.${CMAKE_VERSION_BUILD}.tar.gz \
+    && tar -xzvf cmake-${CMAKE_VERSION}.${CMAKE_VERSION_BUILD}.tar.gz \
+    && cd cmake-${CMAKE_VERSION}.${CMAKE_VERSION_BUILD} \
+    && ./bootstrap \
+    && make -j$(nproc) \
+    && sudo make install \
+    && rm -Rf $CMAKE_BUILD_DIR \
+    && cd \
+    && pip install -U -i $INTEL_PYPI_URL --extra-index-url $BASE_PYPI_URL "numpy==${INTEL_NUMPY_VERSION}" "numba${INTEL_NUMBA_VERSION}" "scipy${INTEL_SCIPY_VERSION}" wheel cython scikit-build
```

This should resolve [the problem mentioned by @ogrisel](https://github.com/ogrisel/sklearn-numba-dpex/pull/1/files#r914945843)

```suggestion
    && pip install -U -i $INTEL_PYPI_URL --extra-index-url $BASE_PYPI_URL "numpy${INTEL_NUMPY_VERSION}" "numba${INTEL_NUMBA_VERSION}" "scipy${INTEL_SCIPY_VERSION}" wheel cython scikit-build
```